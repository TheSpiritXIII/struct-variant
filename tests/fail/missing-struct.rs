#[path="../common/mod.rs"]
mod common;

use common::*;
use struct_variant::struct_variant;

#[struct_variant]
enum ShapeEnumNoBound {
	Circle,
	Rectangle,
	DoesNotExistNoBound // Will error.
}

#[struct_variant(Shape)]
enum ShapeEnumShapeBound {
	Circle,
	Rectangle,
	DoesNotExistShapeBound // Will error.
}

fn main() {}
