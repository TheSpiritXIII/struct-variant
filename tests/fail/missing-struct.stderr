error[E0412]: cannot find type `DoesNotExistNoBound` in this scope
  --> $DIR/missing-struct.rs:11:2
   |
11 |     DoesNotExistNoBound // Will error.
   |     ^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
help: there is an enum variant `crate::ShapeEnumNoBound::DoesNotExistNoBound`; try using the variant's enum
   |
11 |     crate::ShapeEnumNoBound // Will error.
   |     ^^^^^^^^^^^^^^^^^^^^^^^

error[E0412]: cannot find type `DoesNotExistShapeBound` in this scope
  --> $DIR/missing-struct.rs:18:2
   |
18 |     DoesNotExistShapeBound // Will error.
   |     ^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
help: there is an enum variant `crate::ShapeEnumShapeBound::DoesNotExistShapeBound`; try using the variant's enum
   |
18 |     crate::ShapeEnumShapeBound // Will error.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
